libs/go/aealite/protocols/dialogue.go

```diff
@@ -0,0 +1,142 @@
+package protocols
+
+import (
+	"fmt"
+	"crypto/rand"
+	"encoding/hex"
+)
+
+type Address string
+type Performative []string
+type SomeMessageType string
+
+const(
+	NONCE_BYTES_NB = 32
+)
+
+// type Dialogue struct {
+// 	_dialogues_storage
+// 	_dialogues_stats
+// 	_keep_terminal_state_dialogues
+// 	_message_class
+// 	_dialogue_class
+// 	_role_from_first_message
+// 	_self_address
+// 	_outgoing_messages
+// 	_incoming_messages
+// }
+
+// TODO
+// Define RoleType and figure out how to find role
+
+type Label struct {
+	dialogueReference [2]string
+	dialogueOpponentAddress Address
+	dialogueStarterAddress Address
+}
+
+type Dialogue struct {
+	dialogueLabel Label
+	dialogueMessage *InitialMessage
+	selfAddress Address
+	// role RoleType
+}
+
+var dialogueStorage map[Address][]Label
+
+type InitialMessage struct {
+	dialogueReference [2]string
+	message_id int
+	target int
+	performative []string
+	message SomeMessageType
+	to Address
+	sender Address
+}
+
+func create(selfAddress Address, counterParty Address, performative Performative, message SomeMessageType) *Dialogue {
```

Comment:
first argument should be the Dialogues or DialogueCollection

Reply:
Drop the message and instead, we construct based on the message type assigned to the given dialogue.